lib/vrl/stdlib/src/encode_logfmt.rs

```diff
@@ -0,0 +1,234 @@
+use vrl::prelude::*;
+
+mod logfmt {
+    use std::collections::{BTreeMap, HashSet};
+    use std::fmt::{self, Write};
+    use std::result::Result;
+
+    use vrl::prelude::*;
+
+    fn encode_string(output: &mut String, str: &str) -> fmt::Result {
+        let needs_quotting = str.find(' ').is_some();
+        if needs_quotting {
+            output.write_char('"')?;
```

Comment:
I think it is ok to use `expect()` here and avoid returning a `Result` given that writes to a `String` can never fail:

https://doc.rust-lang.org/src/alloc/string.rs.html#2418-2430

This would allow you to change the function to infallible.